pennylane/labs/dla/lie_closure_dense.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A function to compute the Lie closure of a set of operators"""
+# pylint: disable=too-many-arguments
+
+import warnings
+from collections.abc import Iterable
+from typing import Union
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import Operator
+from pennylane.pauli import PauliSentence, PauliWord
+
+
+def _hermitian_basis(matrices: Iterable[np.ndarray], tol: float = None, subbasis_length: int = 0):
+    """Find a linearly independent basis of a list of (skew-) Hermitian matrices
+
+    Args:
+        matrices (Iterable[numpy.ndarray]): A list of Hermitian matrices.
+        tol (float): Tolerance for linear dependence check. Defaults to ``1e-10``.
+
+    Returns:
+        np.ndarray: Stacked array of linearly independent basis matrices.
+
+    Raises:
+        ValueError: If not all input matrices are (skew-) Hermitian.
```

Comment:
As we go through extra effort to make sure not to modify the input matrices, maybe one could put a brief comment into the docstring (even though it is a private method, just for future dev).

Reply:
Not sure I completely follow, what would you describe here?

Reply:
Maybe just mention that the input matrices are not (supposed) to be modified. Then future modifications of this function have a contract they have to adhere to, and functions using this method know what they can rely on. It's a very optional suggestion :)

Reply:
Ah gotcha, I think it's a good idea!

Reply:
✅

Reply:
Haha nice! 
So I meant that the input `matrices` is not being changed by the function. However, the change that you made is way more relevant to users, and I'm quite happy with what the misunderstanding resulted in :D Thanks!

(no further action required, just wanted to comment).